### PR TITLE
feat(minify): add missing `toplevel`, `nameCache`, `keep_classnames`, `keep_fnames` and `safari10` options (`uglifyOptions`)

### PR DESCRIPTION
**Pull request by evilebottnawi**
Also sort options as uglify docs.

Add all options and tests. In next PR i try to implement internal cache for `nameCache` for increase perf :+1:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -160,22 +160,28 @@ Number of concurrent runs.
 
 |Name|Type|Default|Description|
 |:--:|:--:|:-----:|:----------|
-|**`ie8`**|`{Boolean}`|`false`|Enable IE8 Support|
 |**`ecma`**|`{Number}`|`undefined`|Supported ECMAScript Version (`5`, `6`, `7` or `8`). Affects `parse`, `compress` && `output` options|
+|**`warnings`**|`{Boolean}`|`false`|Display Warnings|
```
Comment (michael-ciniawsky) [inline README.md]:
```
**[warnings](#warnings)**
```

(if there is a link, otherwise ignore)

File: README.md [hunk 2]
```diff
@@ -160,22 +160,28 @@ Number of concurrent runs.
...
 |**[`output`](https://github.com/mishoo/UglifyJS2/tree/harmony#output-options)**|`{Object}`|`{}`|Additional Output Options (The defaults are optimized for best compression)|
-|**[`compress`](https://github.com/mishoo/UglifyJS2/tree/harmony#compress-options)**|`{Boolean\|Object}`|`true`|Additional Compress Options|
-|**`warnings`**|`{Boolean}`|`false`|Display Warnings|
+|**`toplevel`**|`{Boolean}`|`false`|Enable top level variable and function name mangling and to drop unused variables and functions|
+|**`nameCache`**|`{Object}`|`null`|Enable cache of mangled variable and property names across multiple invocations|
+|**`ie8`**|`{Boolean}`|`false`|Enable IE8 Support|
+|**`keep_classnames`**|`{Boolean}`|`undefined`|Enable prevent discarding or mangling of class names|
+|**`keep_fnames`**|`{Boolean}`|`false`| Enable prevent discarding or mangling of function names. Useful for code relying on `Function.prototype.name`. If the top level minify option `keep_classnames` is `undefined` it will be overriden with the value of the top level minify option `keep_fnames`.|
```
Comment (michael-ciniawsky) [inline README.md]:
```diff
- ... `keep_fnames`.
+ ... `keep_fnames` 
```

```diff
- .
```

No  'final'  `.` in `Description`

File: README.md [hunk 3]
```diff
@@ -160,22 +160,28 @@ Number of concurrent runs.
...
+|**`safari10`**|`{Boolean}`|`false`|Enable work around Safari 10/11 bugs in loop scoping and `await`.|
```
Comment (michael-ciniawsky) [inline README.md]:
```diff
- .
```

File: README.md (hunk 1 above)
Comment (evilebottnawi) [reply README.md]: No link

**Comment by evilebottnawi** [comment]
@michael-ciniawsky done :+1:

**Comment by michael-ciniawsky** [comment]
Released in [`v1.2.0`](https://github.com/webpack-contrib/uglifyjs-webpack-plugin/releases/tag/v1.2.0) 🎉

**Pull request merged**